test/unit/runner.spec.js

```diff
@@ -496,6 +497,62 @@ describe('Runner', function() {
         runner.failHook(hook, err);
       });
     });
+
+    describe('hugeStackTrace', function() {
+      beforeEach(function() {
+        if (path.sep !== '/') {
+          this.skip();
+        }
+      });
+
+      it('should not hang if the error message is ridiculously long single line', function(done) {
+        var hook = new Hook();
+        hook.parent = suite;
+        var data = [];
+        // mock a long message
+        for (var i = 0; i < 10000; i++) data[i] = {a: 1};
+        var message = JSON.stringify(data);
```

Comment:
```js
function genSingleLineMessage() {
  var data = [];
  for (var i = 0; i < 10000; i++) data[i] = {a: 1};
  return JSON.stringify(data);
}
var message = genSingleLineMessage();
```

Reply:
I think the test case title is self explanatory enough. The test case body isn't that long that needs this level of refactor. Can I pass on this one?

Reply:
Each function would live outside the corresponding `it()` replacing the _how_ with _what_ was generated. Prefer self-documenting code (especially considering the number of questions this small PR has already brought up). But whatever...

Reply:
Let's just discuss in this thread. I believe the changes you proposed are all quite similar. Extracting three lines into a function, breaking one line into multiple or joining multiple into one, these are all personal preferences. This repo has already configured quite strict (far too strict in my opinion) rules, and I've done the changes to comply with them already. 

And about *the number of questions this small PR has already brought up* , at least half of them were because you guys didn't read my PR description or the code carefully to understand in the first place. A week has passed since I requested this PR, four reviewers have reviewed the code, more than 55 comments were made, the real change that actually matters is to remove five characters `\(?.+`, and guess what, it is still not merged.

I understand that people get busy and stuff, but I'm done. Take it or leave it.

Reply:
> And about _the number of questions this small PR has already brought up_ , at least half of 
> them were because you guys didn't read my PR description or the code carefully to 
> understand in the first place. A week has passed since I requested this PR, four reviewers 
> have reviewed the code, more than 55 comments were made, the real change that actually 
> matters is to remove five characters `\(?.+`, and guess what, it is still not merged.

I _read_ your PR description. Problem is that you're making the assumption we're extremely familiar with the _entirety_ of the codebase. **Most of us aren't.** As such, we have to learn what the original code does, figure out what changes you made, whether it could have been done differently, what is tested (or wasn't), and what possible breakage could come. We also have to maintain it afterwards, typically only having gained cursory knowledge of the original problem. 
And all that takes unpaid time away from our families... Sorry we're too slow for your taste.